Content.Server/AlertLevel/AlertLevelSystem.cs

```diff
@@ -139,8 +141,9 @@ public string GetDefaultLevel(Entity<AlertLevelComponent?> station)
     /// <param name="announce">Say the alert level's announcement.</param>
     /// <param name="force">Force the alert change. This applies if the alert level is not selectable or not.</param>
     /// <param name="locked">Will it be possible to change level by crew.</param>
+    /// <param name="caller">Entity that called this method, for logging purposes.</param>
     public void SetLevel(EntityUid station, string level, bool playSound, bool announce, bool force = false,
-        bool locked = false, MetaDataComponent? dataComponent = null, AlertLevelComponent? component = null)
+        bool locked = false, EntityUid? caller = null, MetaDataComponent? dataComponent = null, AlertLevelComponent? component = null)
```

Comment:
```suggestion
        bool locked = false, EntityUid? user= null, MetaDataComponent? dataComponent = null, AlertLevelComponent? component = null)
```
this should be the player changing the alert level, not the device

Reply:
> this should be the player changing the alert level, not the device

how would I even get the player from something, like, an entity that has AlertLevelChangeOnTrigger that got triggered by a TriggeOnSignal?

i think it would be easier to list what entity caused the alert level to change, so an admin can find it if needed.